cmd/sonarExecuteScan.go

```diff
@@ -84,6 +97,21 @@ func runSonar(config sonarExecuteScanOptions, client piperhttp.Downloader, runne
 		// handleArtifactVersion is reused from cmd/protecodeExecuteScan.go
 		sonar.addOption("sonar.projectVersion=" + handleArtifactVersion(config.ProjectVersion))
 	}
+	if len(config.ProjectKey) > 0 {
+		sonar.addOption("sonar.projectKey=" + config.ProjectKey)
+	}
+	if len(config.M2Path) > 0 && !SliceUtils.ContainsStringPart(config.Options, javaLibraries) {
```

Comment:
This could interfere with `sonar.java.libraries` defined in the `sonar-project.properties` file I guess.

Reply:
Solved by requiring activation via new parameter `inferJavaLibraries`.